src/format/mod.rs

```diff
@@ -494,10 +494,17 @@ pub fn format<'a, I>(w: &mut fmt::Formatter, date: Option<&NaiveDate>, time: Opt
                     RFC2822 => // same to `%a, %e %b %Y %H:%M:%S %z`
                         if let (Some(d), Some(t), Some(&(_, off))) = (date, time, off) {
                             let sec = t.second() + t.nanosecond() / 1_000_000_000;
-                            try!(write!(w, "{}, {:2} {} {:04} {:02}:{:02}:{:02} ",
```

Comment:
Also, Is `Tue, 03 Feb 2015 23:16:09 +0500` be better ?

Reply:
Since this number is guaranteed to be between 1 and 31 inclusive, it should be fine to just remove the the width specifier -- just use `{}`. I think that's what should be done here.

If you ever need to deal with something like this again, though, you can use the [`$` width specifier to supply the width as a separate argument](https://doc.rust-lang.org/std/fmt/#width). In this case it would be something like:

```rust
let wwwwidth = if d.day() < 10 { 1 } else { 2 };
try!(write!(w, "{}, {:$} {} {:04} {:02}:{:02}:{:02} ",
    SHORT_WEEKDAYS[d.weekday().num_days_from_monday() as usize],
    d.day(), wwwwidth, SHORT_MONTHS[d.month0() as usize], d.year(),
    t.hour(), t.minute(), sec));
```

Again, I don't think we should do that here, but it's kind of nifty.

Reply:
Done.